lib/prop/leaky_bucket_strategy.rb

```diff
@@ -8,9 +8,10 @@ class << self
       def counter(cache_key, options)
         bucket = Prop::Limiter.cache.read(cache_key) || zero_counter
         now = Time.now.to_i
-        leak_amount = (now - bucket.fetch(:last_updated)) / options.fetch(:interval) * options.fetch(:threshold)
+        leak_rate = (now - bucket.fetch(:last_updated)) / options.fetch(:interval).to_f
+        leak_amount =  leak_rate * options.fetch(:threshold)
 
-        bucket[:bucket] = [bucket.fetch(:bucket) - leak_amount, 0].max
+        bucket[:bucket] = [(bucket.fetch(:bucket) - leak_amount).to_i, 0].max
```

Comment:
now it can still be 0 ? <-> should we get rid of 0 and make it `, 1].max` ?

Reply:
I think leaving 0 here is fine. `max` with 0 resets the bucket if the throttle hasn't been checked within the interval.